VisualScratchSpace/ViewModel/CheckBoxImporterVM.cs

```diff
@@ -0,0 +1,214 @@
+    public ObservableCollection<Simulation> SimulationsComboBox { get; set; }
+    public ObservableCollection<CheckableItem> AlternativesCheckBox { get; set; }
+    public ObservableCollection<CheckableItem> HazardTimesCheckBox { get; set; }
+
+    private int _plotIndex = 0;
+    private List<LifeLossFunction> _lifeLossFunctions = new();
+
+    [ObservableProperty]
+    private string _selectedPath = "";
+    [ObservableProperty]
+    private string _selectedSimulation = "";
+    [ObservableProperty]
+    private string _selectedPointsPath = @"C:\FDA_Test_Data\WKS20230525\WKS20230525\index_points.shp";
+    [ObservableProperty]
+    private string _selectedSummarySetPath = @"C:\FDA_Test_Data\WKS20230525\WKS20230525\summary_set.shp";
+    [ObservableProperty]
+    private string _selectedHydraulicsFolder = @"C:\FDA_Test_Data\WKS20230525\WKS20230525\Hydraulic_Data";
+
+    public CheckBoxImporterVM()
+    {
+        SimulationsComboBox = [];
+        AlternativesCheckBox = [];
+        HazardTimesCheckBox = [];
+        MyModel = new PlotModel();
+    }
+
+    /// <summary>
+    /// Import the selected simulation parametes to create associated functions and plots
+    /// </summary>
+    [RelayCommand]
+    public void Import()
+    {
+        if (SelectedPath.IsNullOrWhiteSpace() || SelectedSimulation.IsNullOrWhiteSpace() || SelectedHydraulicsFolder.IsNullOrWhiteSpace()) return;
+        if (AlternativesCheckBox.IsEmpty() || HazardTimesCheckBox.IsEmpty()) return;
+
+        // determine which parameters have been selected
+        Simulation currentSimulation = new(SelectedSimulation, SelectedHydraulicsFolder);
+        foreach (CheckableItem a in AlternativesCheckBox)
+            if (a.IsChecked) currentSimulation.Alternatives.Add(a.Name);
+        foreach (CheckableItem h in HazardTimesCheckBox)
+            if (h.IsChecked) currentSimulation.HazardTimes.Add(h.Value);
+
+        // send those parameters to the a generator
+        LifeLossFunctionGenerator generator = new(SelectedPath, currentSimulation);
+
+        if (SelectedSummarySetPath.IsNullOrWhiteSpace() || SelectedPointsPath.IsNullOrWhiteSpace()) return;
+
+        // generate associated functions for the selected parameters and plot the first one
+        _lifeLossFunctions.Clear();
+        _lifeLossFunctions = generator.CreateLifeLossFunctions(SelectedSummarySetPath, SelectedPointsPath);
+        _plotIndex = 0;
+        ChangePlot(_plotIndex);
+    }
+
+    /// <summary>
+    /// Cycle through all plots and loop back to start once the end is passed
+    /// </summary>
+    [RelayCommand]
+    public void NextPlot()
+    {
+        if (_lifeLossFunctions.IsEmpty()) return;
+        _plotIndex = (_plotIndex + 1) % _lifeLossFunctions.Count;
+        ChangePlot(_plotIndex);
+    }
+
+    [RelayCommand]
+    public void SavePlots()
+    {
+        string dbpath = @"C:\FDA_Test_Data\WKS20230525\WKS20230525\save-test.db";
+        LifeLossPlotSaver ps = new(dbpath);
```

Comment:
Nomenclature thing here. We don't actually save plots, because plots are pictures of data. We actually save the backing data, which I might call a relationship, or a function. Some preference for function. Maybe LifeLossFunctionSaver?